hooks/post/30-upgrade.rb

```diff
@@ -171,30 +80,9 @@ def fail_and_exit(message)
     if Kafo::Helpers.module_enabled?(@kafo, 'katello')
       upgrade_step :db_seed, :run_always => true
 
-      upgrade_step :correct_repositories, :long_running => true, :run_always => true
-      upgrade_step :correct_puppet_environments, :long_running => true, :run_always => true
-      upgrade_step :clean_backend_objects, :long_running => true, :run_always => true
-
-      upgrade_step :import_package_groups, :long_running => true
-      upgrade_step :import_rpms, :long_running => true
-      upgrade_step :import_distributions, :long_running => true
-      upgrade_step :import_puppet_modules, :long_running => true
-      upgrade_step :import_subscriptions, :long_running => true
-      upgrade_step :import_product_content, :long_running => true
-      upgrade_step :add_export_distributor, :long_running => true
-      upgrade_step :remove_docker_v1_content, :long_running => true
-      upgrade_step :update_puppet_repository_distributors, :long_running => true
-      upgrade_step :update_subscription_facet_backend_data, :long_running => true
       upgrade_step :remove_event_queue
-      upgrade_step :set_virt_who_on_pools, :long_running => true
-      upgrade_step :remove_unused_products, :long_running => true
-      upgrade_step :create_host_subscription_associations, :long_running => true
-      upgrade_step :reindex_docker_tags, :long_running => true
-      upgrade_step :republish_file_repos, :long_running => true
-      upgrade_step :import_backend_consumer_attributes, :long_running => true
-      upgrade_step :remove_registration_tasks
       upgrade_step :remove_legacy_mongo
-      upgrade_step :ensure_ks_repos_are_bootable, :long_running => true
+      upgrade_tasks
```

Comment:
@ekohl I wasn't sure if this is how you expected this to be run, or if we could re-use a hook in foreman's installer (and i guess add a --upgrade flag to it?).

Reply:
Since this is in Foreman core I think this should be done if the Foreman module is enabled rather than just katello. For database migrations we also have a setting that indicates a migration is needed. The same goes for seeds. See https://github.com/theforeman/puppet-foreman/blob/b937165001aac788fbafef0422984a59edaeba87/manifests/database.pp#L33-L46. I think this would be good thing for the upgrade as well. It could even be done in pure Puppet similar to `db:migrate` and `db:seed` (yes, in katello-installer we run migration and seeding twice now) but it'd be fine to do that in a phase 2.

Reply:
@ekohl  for this 'first phase' should this live in foreman-installer?  Or are you just saying to leave it here and check for the foreman module.

Reply:
It's fine to let it live here initially.